src/ir/item.rs

```diff
@@ -935,23 +936,23 @@ impl Item {
         let cc = &ctx.options().codegen_config;
         match *self.kind() {
             ItemKind::Module(..) => true,
-            ItemKind::Var(_) => cc.vars,
-            ItemKind::Type(_) => cc.types,
-            ItemKind::Function(ref f) => {
-                match f.kind() {
-                    FunctionKind::Function => cc.functions,
-                    FunctionKind::Method(MethodKind::Constructor) => {
-                        cc.constructors
-                    }
-                    FunctionKind::Method(MethodKind::Destructor) |
-                    FunctionKind::Method(MethodKind::VirtualDestructor { .. }) => {
-                        cc.destructors
-                    }
-                    FunctionKind::Method(MethodKind::Static) |
-                    FunctionKind::Method(MethodKind::Normal) |
-                    FunctionKind::Method(MethodKind::Virtual { .. }) => cc.methods,
+            ItemKind::Var(_) => cc.contains(CodegenConfig::VARS),
+            ItemKind::Type(_) => cc.contains(CodegenConfig::TYPES),
+            ItemKind::Function(ref f) => match f.kind() {
+                FunctionKind::Function => cc.contains(CodegenConfig::FUNCTIONS),
+                FunctionKind::Method(MethodKind::Constructor) => {
+                    cc.contains(CodegenConfig::CONSTRUCTORS)
                 }
-            }
+                FunctionKind::Method(MethodKind::Destructor)
+                | FunctionKind::Method(MethodKind::VirtualDestructor {
+                    ..
+                }) => cc.contains(CodegenConfig::DESTRUCTORS),
+                FunctionKind::Method(MethodKind::Static)
+                | FunctionKind::Method(MethodKind::Normal)
```

Comment:
The reformat here seems kind of unfortunate, I don' think we use the pipe at the beginning of the line in other places?

Reply:
This is the output from rustfmt. I put it back the way it was before.